Add Barbican adoption with Proteccio HSM integration documentation (#1058)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_adopting-key-manager-service-with-proteccio-hsm.adoc

* The PKCS11 crypto plugin is available alongside `simple_crypto` for new secret storage.
* HSM functionality is verified and operational.

[NOTE]
====
This procedure is specifically for environments that use Proteccio HSM. For standard {key_manager} adoption using `simple_crypto`, see xref:adopting-the-key-manager-service_{context}[Adopting the {key_manager}].

The enhanced Barbican adoption role supports HSM configuration through a simple boolean flag. This approach integrates seamlessly with the standard data-plane-adoption framework while providing HSM support.
====

.Prerequisites

* You have a running {OpenStackPreviousInstaller} environment with Proteccio HSM integration (the source cloud).
* You have a Single Node OpenShift or OpenShift Local running in the {rhocp_long} cluster.
* You have SSH access to the source {OpenStackPreviousInstaller} undercloud and controller nodes.
* You have configured HSM variables in your adoption configuration files.
* Custom Barbican container images with the Proteccio client libraries are available in your registry.

[IMPORTANT]
====

> The HSM adoption process requires proper configuration of HSM-related variables. The adoption role automatically creates the required Kubernetes secrets (`hsm-login` and `proteccio-data`) when `barbican_hsm_enabled` is set to `true`. Ensure that:

Reviewer: ```suggestion
The HSM adoption process requires proper configuration of HSM-related variables. The adoption role automatically creates the required Kubernetes secrets (`hsm-login` and `proteccio-data`) when `barbican_hsm_enabled` is set to `true`. Ensure that the following items are true:
```

* All HSM-related variables are properly set in your configuration files
* The Proteccio client ISO, certificates, and configuration files are accessible from the configured URLs

> * Custom Barbican images with Proteccio client are built and available in your container registry

Reviewer: ```suggestion
* All HSM-related variables are properly set in your configuration files.
* The Proteccio client ISO, certificates, and configuration files are accessible from the configured URLs.
* Custom {key_manager}  images with Proteccio client are built and available in your container registry.
```

Reviewer: Can this list be merged with the Prerequisites?

cifmw_barbican_proteccio_library_path: "/usr/lib64/libnethsm.so"
cifmw_barbican_proteccio_key_wrap_mechanism: "CKM_AES_CBC_PAD"

# HSM client sources (URLs to download Proteccio client files)
cifmw_hsm_proteccio_client_src: "https://your-server/path/to/Proteccio3.06.05.iso"
cifmw_hsm_proteccio_conf_src: "https://your-server/path/to/proteccio.rc"
cifmw_hsm_proteccio_client_crt_src: "https://your-server/path/to/client.crt"
cifmw_hsm_proteccio_client_key_src: "https://your-server/path/to/client.key"
cifmw_hsm_proteccio_server_crt_src:
  - "https://your-server/path/to/hsm_server.CRT"
----

. Run the data-plane-adoption tests with HSM support enabled. The adoption process will:
+
--
* Extract the simple crypto KEK from the source configuration
* Create the required HSM secrets (`hsm-login` and `proteccio-data`) in the target namespace
* Deploy Barbican with HSM-enabled configuration using the PKCS#11 plugin
* Verify HSM functionality and secret migration
--

.Verification

. Ensure that the {identity_service_first_ref} endpoints are defined and are pointing to the control plane FQDNs:
+
----
$ openstack endpoint list | grep key-manager
----

> . Ensure that Barbican API service is registered in the {identity_service}:

Reviewer: ```suggestion
. Ensure that the Barbican API service is registered in the {identity_service}:
```
